uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/interaction/b2c/GoogleLoginComponentHandler.java

```diff
@@ -0,0 +1,108 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.client.ui.automation.interaction.b2c;
+
+import android.widget.EditText;
+
+import androidx.annotation.NonNull;
+import androidx.test.uiautomator.UiObject;
+import androidx.test.uiautomator.UiObjectNotFoundException;
+import androidx.test.uiautomator.UiSelector;
+
+import com.microsoft.identity.client.ui.automation.interaction.ILoginComponentHandler;
+import com.microsoft.identity.client.ui.automation.utils.UiAutomatorUtils;
+
+import org.junit.Assert;
+
+/**
+ * A login component handler for Google IdP
+ */
+public class GoogleLoginComponentHandler implements ILoginComponentHandler {
+
+    @Override
+    public void handleEmailField(@NonNull final String username) {
+        UiAutomatorUtils.handleInput("identifierId", username);
+        handleNextButton();
+    }
+
+    @Override
+    public void handlePasswordField(@NonNull final String password) {
+        UiAutomatorUtils.handleInput("password", password);
+        final UiObject passwordBox = UiAutomatorUtils.obtainUiObjectWithResourceId("password");
```

Comment:
why do we have to do 51-58 here (but we don't have to in FB's case?)

Reply:
FB, thankfully declared a resource id for each of the edit text fields, however, google didn't. In this case, Google instead has a resource id assigned to parent outer level password box that contains a label for the inner password edit text field. The actually inner edit text field doesn't any resource id....hence we need to do this logic to find the actual element for the password field.